File: c/1297.c
#include <stdio.h>
 
int main()
{
    int t, max, min, i, num;
 
    scanf("%d", &t);
    for(i=0;i<t;i++){
        scanf("%d", &num);
        if(num%2==1)
            printf("0 0\n");
        else{
            max=num/2;
            min=num/4;
            if(num%4==2)
                min++;
            printf("%d %d\n", min, max);
        }
    }
    return 0;
}

File: java/1297.java
import java.util.*;
 
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
 
        int t, max, min, i, num;
 
        t = sc.nextInt();
        for(i=0;i<t;i++){
            num = sc.nextInt();
            if(num%2==1)
                System.out.printf("0 0\n");
            else{
                max=num/2;
                min=num/4;
                if(num%4==2)
                    min++;
                System.out.printf("%d %d\n", min, max);
            }
        }
    }
}